src/MizarData/include/MizarData/BaselineOrComparison.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_DATA_BASELINE_OR_COMPARISON_H_
+#define MIZAR_DATA_BASELINE_OR_COMPARISON_H_
+
+#include <type_traits>
+#include <utility>
+
+namespace orbit_mizar_data {
+
+template <typename T, typename U>
+using EnableIfUConvertibleToT = std::enable_if_t<std::is_convertible_v<U, T>>;
+
+template <typename T>
+class BaselineOrComparison {
+ public:
+  ~BaselineOrComparison() = default;
+
+  const T* operator->() const { return &value_; }
+  T* operator->() { return &value_; }
+
+  const T& operator*() const { return value_; }
+  T& operator*() { return value_; }
+
+ protected:
+  template <typename U, typename = EnableIfUConvertibleToT<T, U>>
+  explicit BaselineOrComparison(U&& value) : value_(std::forward<T>(value)) {}
+  BaselineOrComparison(BaselineOrComparison&& other) = default;
+
+  template <typename... Args>
+  explicit BaselineOrComparison(std::in_place_t /*_*/, Args&&... args)
```

Comment:
Use the argument name instead of `_`.

Reply:
What would it be for the in-place flag? 

But I see your point, I often use underscore instead of the name when the name is not used. IMO this increases consistency. When we use structural binding and we ignore one of the elements, underscore (again, IMO) looks clearer (`const [_, key] : map`).

Reply:
> What would it be for the in-place flag?

`in_place`?

> When we use structural binding and we ignore one of the elements, underscore (again, IMO) looks clearer.

We had this exact discussion in the past and I don't really want to start over. Long story short: underscore in many languages is used for that purpose and is even supported by the language. Not in C++, where: anything with a leading underscore is subject to some rules; you could have conflicts in the same scope. For structured bindings, we had settled on the `unused_` prefix, `[unused_key, value] : map` for example.

Reply:
Got it, thanks. But in the case of `std::in_place_t` I'll just omit the name to match STL.